File: src/main/java/noteedit/service/NoteService.java
package com.gliese.noteedit.service;

import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper;
import com.gliese.noteedit.entity.Note;
import com.gliese.noteedit.mapper.NoteMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class NoteService {
    @Autowired
    private NoteMapper noteMapper;

    public void saveNote(Note note) {
        QueryWrapper<Note> queryWrapper = new QueryWrapper<>();
        queryWrapper.eq("file_id", note.getFileId());
        Note getNote = noteMapper.selectOne(queryWrapper);
        if(getNote == null) { // * 新笔记
            noteMapper.insert(note);
        } else { // * 更新笔记
            note.setId(getNote.getId());
            noteMapper.updateById(note);
        }
    }

    public Note getNoteByFile(Long fileId) {
        QueryWrapper<Note> queryWrapper = new QueryWrapper<>();
        queryWrapper.eq("file_id", fileId);
        return noteMapper.selectOne(queryWrapper);
    }
}

File: src/main/java/noteedit/service/NoteService.kt
package noteedit.service

import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper
import noteedit.entity.Note
import noteedit.mapper.NoteMapper
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
open class NoteService {
    @Autowired
    private lateinit var noteMapper: NoteMapper

    open fun save(note: Note): Boolean {
        return noteMapper.selectOne(
            QueryWrapper<Note>().eq("file_id", note.fileId)
        )?.let {
            // update
            note.id = it.id
            noteMapper.updateById(note) == 1
        } ?: (noteMapper.insert(note) == 1) // insert
    }

    open fun getByFileId(fileId: Long): Note? {
        return noteMapper.selectOne(
            QueryWrapper<Note>().eq("file_id", fileId)
        )
    }
}